glean-core/src/lib.rs

```diff
@@ -869,6 +869,18 @@ pub fn glean_test_get_experiment_data(experiment_id: String) -> Option<RecordedE
     core::with_glean(|glean| glean.test_get_experiment_data(experiment_id.to_owned()))
 }
 
+/// Set an experimentation identifier dynamically.
+///
+/// Note: it's probably a good idea to unenroll from any experiments when identifiers change.
+pub fn glean_set_experimentation_id(experimentation_id: String) {
+    core::with_glean(|glean| {
+        glean
+            .additional_metrics
+            .experimentation_id
+            .set_sync(glean, experimentation_id.to_string());
```

Comment:
This is a normal metric and thus this would cause IO on the main thread, because this is not dispatched like other metrics. Is that intended?

Reply:
Nope, this doesn't need to do IO on the main thread, I'll change this to use the proper `with_glean`